buildSrc/src/main/java/org/opensearch/gradle/transform/SymbolicLinkPreservingUntarTransform.java

```diff
@@ -57,42 +57,37 @@ public void unpack(File tarFile, File targetDir) throws IOException {
             .info("Unpacking " + tarFile.getName() + " using " + SymbolicLinkPreservingUntarTransform.class.getSimpleName() + ".");
         Function<String, Path> pathModifier = pathResolver();
 
-        try (
-            FileInputStream fis = new FileInputStream(tarFile);
-            GzipCompressorInputStream gzip = new GzipCompressorInputStream(fis);
-            TarArchiveInputStream tar = new TarArchiveInputStream(gzip)
-        ) {
-            final Path destinationPath = targetDir.toPath();
-            TarArchiveEntry entry = tar.getNextTarEntry();
-            while (entry != null) {
-                final Path relativePath = pathModifier.apply(entry.getName());
-                if (relativePath == null || relativePath.getFileName().equals(CURRENT_DIR_PATH)) {
-                    entry = tar.getNextTarEntry();
-                    continue;
-                }
+        TarArchiveInputStream tar = new TarArchiveInputStream(new GzipCompressorInputStream(new FileInputStream(tarFile)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *tar*. Execution paths that do not contain closure statements were detected. Ensure *tar* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).